File: passwordGenerator.cpp
//Equivalent C++ code

#include<iostream>
#include <ctime>
#include<conio.h>

using namespace std;

int main(){
  string numbers = "0123456789";
  string smallLetters = "abcdefghijklmnopqrstuvwxyz";
  string capitalLetters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
  string symbols = "`~!@#$%^&*()_+-={}[]:'.,><?/";
  int length;
  
  string password = "";
  string characters = "";
  char choice;
 
  srand(time(0));

  cout<<endl<<"Enter Length of password ";
  cin>>length;
  cout<<endl<<"Specialized Password(s) or Quick Password(any)?";
  cin>>choice;
  if(choice == 's'){
    cout<<endl<<"Include Numbers?(y/n) ";
    cin>>choice;
    if(choice == 'y'){
      characters += numbers;
    }

    cout<<endl<<"Include SmallLetters?(y/n) ";
    cin>>choice;
    if(choice == 'y'){
      characters += smallLetters;
    }
    
    cout<<endl<<"Include CapitalLetters?(y/n) ";
    cin>>choice;
    if(choice == 'y'){
      characters += capitalLetters;
    }
    
    cout<<endl<<"Include Symbols?(y/n) ";
    cin>>choice;
    if(choice == 'y'){
      characters += symbols;
    }

  }else{
    characters = numbers + smallLetters + capitalLetters + symbols;
  }

  for(int i=0;i<length;i++){
    password += characters[rand() % characters.size()];
  }

  cout<<endl<<endl<<"------------------"
      <<endl<<"Your Password = "<<password;

  getch();
}

File: passwordGenerator.py
# Equivalent Python Code

import random,string

numbers = '0123456789'
smallLetter = "abcdefghijklmnopqrstuvwxyz"
capitalLetter = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
symbols = '`~!@#$%^&*()_+=-[]{};:",./?'

def generateCharacters(Choice):
  return random.choice(Choice)

def generatePassword():
  password = ""
  length = int(input("Enter the length of Password "))
  passwordType = input("Specilized Password(s) or Quick Password(any)?");

  print("--------------------------");

  if passwordType == 's' or passwordType == 'S':
    numberY,smallLetterY,capitalLetterY,symbolsY = input("Should it contain Numbers?(y/n)"),input("Small Letters? "),input("Capital Letters? "),input("Symbols? ")
  else:
    numberY,smallLetterY,capitalLetterY,symbolsY = 'y','y','y','y'
  
  print("--------------------------");

  for i in range(0,length):
    temp = ""
    if numberY == 'y' or numberY == 'Y':
      temp += generateCharacters(numbers)
    if smallLetterY == 'y' or smallLetterY == 'Y':
      temp += generateCharacters(smallLetter)
    if capitalLetterY == 'y' or capitalLetterY == 'Y':
      temp += generateCharacters(capitalLetter)
    if symbolsY == 'y' or symbolsY == 'Y':
      temp += generateCharacters(symbols)

    password += random.choice(temp)
  
  print(f"Password = {password}")

generatePassword()


# another quick way
print("-------Quick Way-------");

def quickPassword():
  # secure_random = random.SystemRandom()
  alphabet = string.ascii_letters + string.digits + string.punctuation
  # password = "".join(secure_random.choices(alphabet,k=16));
  # print(f"Quick Password = {password}")
  password = "".join(random.sample(alphabet,16))
  print(f"Quick Password = {password}")

quickPassword()